File: threedollar-infrastructure/threedollar-monitoring/src/main/java/com/depromeet/threedollar/infrastructure/monitoring/config/WebSecurityConfig.java
package com.depromeet.threedollar.infrastructure.monitoring.config;

import org.springframework.context.annotation.Configuration;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
import org.springframework.security.config.http.SessionCreationPolicy;

@EnableWebSecurity
@Configuration
public class WebSecurityConfig extends WebSecurityConfigurerAdapter {

    private static final String MONITORING_MATCHERS = "/monitoring/**";

    @Override
    protected void configure(HttpSecurity http) throws Exception {
        http.authorizeRequests()
            .antMatchers(MONITORING_MATCHERS).authenticated()
            .anyRequest().permitAll()
            .and()
            .httpBasic().and()
            .formLogin().disable()
            .logout().disable()
            .csrf().disable()
            .sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS)
            .and()
            .headers()
            .frameOptions().sameOrigin()
            .cacheControl().and()
            .xssProtection().and()
            .httpStrictTransportSecurity().disable();
    }

}

File: threedollar-api-boss/src/main/kotlin/com/depromeet/threedollar/api/boss/config/security/WebSecurityConfig.kt
package com.depromeet.threedollar.api.boss.config.security

import org.springframework.context.annotation.Configuration
import org.springframework.security.config.annotation.web.builders.HttpSecurity
import org.springframework.security.config.annotation.web.builders.WebSecurity
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity
import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter
import org.springframework.security.config.http.SessionCreationPolicy

@EnableWebSecurity
@Configuration
class WebSecurityConfig : WebSecurityConfigurerAdapter() {

    override fun configure(web: WebSecurity) {
        web.ignoring().antMatchers(
            "/h2-console/**",
            "/swagger-resources/**",
            "swagger-ui/**",
            "/v2/api-docs",
            "/webjars/**"
        )
    }

    override fun configure(http: HttpSecurity) {
        http.authorizeRequests()
                .antMatchers("/monitoring/**").authenticated()
                    .anyRequest().permitAll()
            .and()
                .httpBasic()
            .and()
                .sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS)
            .and()
                .csrf().disable()
    }

}
